feat: Display scale as number of browser windows

SuaYoo opened the pull request:
Resolves https://github.com/webrecorder/browsertrix/issues/2048

<!-- Fixes #issue_number -->

### Changes

- Returns `numBrowsers` from API app settings
- Displays scale/crawler instances as "Browser Windows" in workflow editor, workflow detail, and crawl settings
- Fixes info text for edit scale dialog

### Manual testing

1. Log in as crawler
2. Go to "Crawl Workflows" > edit workflow
3. Click "Limits". Verify "Crawler Instances" is not shown
4. Click "Browser Settings". Verify "Browser Windows" is shown
5. Update browser windows and save. Verify workflow saves as expected
6. Run crawl and wait for "Edit Browser Windows" to be enabled
7. Click "Edit Browser Windows". Verify dialog is shown with the correct info text (see screenshot)

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Workflow Editor - Browser Settings | <img width="1057" alt="Screenshot 2024-08-30 at 12 08 12 PM" src="https://github.com/user-attachments/assets/c723fedf-86a1-4866-b343-29a68939cdf4"> |
|  Workflow Detail - Watch Crawl | <img width="504" alt="Screenshot 2024-08-30 at 12 07 47 PM" src="https://github.com/user-attachments/assets/ef7cb63b-78e4-4f4c-b8c0-3c2bc061536e"> |
| Workflow Detail / Archived Item Detail - Crawl Settings | <img width="283" alt="Screenshot 2024-08-30 at 12 07 56 PM" src="https://github.com/user-attachments/assets/e45d831d-d2a1-4936-b642-71b3a9ba8f84"> |



### Follow-ups

There's a bug where the "Edit Browser Settings" button isn't enabled until the crawl is `running`, even though the value can be updated during any active crawl state. Should be addressed with https://github.com/webrecorder/browsertrix/issues/2059

ikreymer committed:
add 'numBrowsers' to /api/settings endpoint, part of #2048

SuaYoo committed:
update field

SuaYoo committed:
update docs

SuaYoo committed:
update info text

SuaYoo committed:
update dialog

SuaYoo committed:
update display value

SuaYoo committed:
update config

SuaYoo committed:
fix when usage

SuaYoo committed:
update dialog name

SuaYoo committed:
fix info text

SuaYoo committed:
update how edit browser disable

SuaYoo committed:
update execution time text

Shrinks99 committed:
Change verb

I think we can use "crawling" here!

Shrinks99 commented on frontend/src/features/crawl-workflows/workflow-editor.ts:
```diff
@@ -1366,6 +1343,38 @@ https://archiveweb.page/images/${"logo.svg"}`}
+            href="https://docs.browsertrix.com/user-guide/workflow-setup/#browser-windows"
+            class="text-blue-600 hover:text-blue-500"
+            target="_blank"
+            >${msg("See caveats")}</a
```
Don't have to do it for this PR, but it would be _really cool_ if docs links we link to in the app opened up in the sidebar!

Shrinks99 committed:
Minor docs reword

Shrinks99 committed:
Fix broken ID links

@SuaYoo IDK if your version of mkdocs is doing this (I think it was released in a newer version, I'm on 1.6), but it will show you in the terminal if you have broken links!  Useful for nailing down stuff like this.

Shrinks99 reviewed (approved):
Looks good to me!  I think this will be a small but helpful bit of de-jargoning! :)

When I was testing earlier, I observed `NaN` values in the workflow settings tab but I'm pretty sure this was a cache issue.  I can't reproduce it and it seems to work as expected.

<img width="313" alt="Screenshot 2024-08-30 at 10 18 46 PM" src="https://github.com/user-attachments/assets/28baddda-f6f6-4e3a-bf75-8d4a7fb01898">

Only one real thought (see comment on the docs link), but can leave for later.

SuaYoo committed:
update notify message

SuaYoo commented on docs/user-guide/overview.md:
```diff
@@ -21,7 +21,7 @@ The crawling panel lists the number of currently running and waiting crawls, as
 
 ??? Question "How is execution time calculated?"
-    Execution time is the total runtime of all [_Crawler Instances_](workflow-setup.md/#crawler-instances) during a crawl. For instance, if _Crawler Instances_ scale is set to 2× and each crawler instance uses 2 minutes of active crawling time, execution time for the crawl will be 4 minutes. Like elapsed time, this is tracked as the crawl runs so changing the _Crawler Instances_ scale while a cra ...
+    Execution time is the total runtime of all [_Crawler Instances_](workflow-setup.md/#browser-windows) during a crawl. For instance, if _Crawler Instances_ scale is set to 2× and each crawler instance uses 2 minutes of active crawling time, execution time for the crawl will be 4 minutes. Like elapsed time, this is tracked as the crawl runs so changing the _Crawler Instances_ scale while a crawl ...
```
@Shrinks99 Should we rewrite this section?

Shrinks99 replied:
gah yes.  It's late here >_<

Shrinks99 replied:
Updated this a bit, it's accurate now though I wonder if it could still be communicated better.

Shrinks99 committed:
Update execution time overview

SuaYoo committed:
update file documentation

tw4l reviewed (approved):
Very nice, and thanks for the thorough job updating copy through the app and in the docs!
